src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java

```diff
@@ -690,49 +694,58 @@ private void buildRespondentCounterClaim(EventHistory.EventHistoryBuilder builde
-        LocalDateTime dateReceived = caseData.getRespondent1TimeExtensionDate();
-        if (dateReceived == null) {
-            return;
+        List<Event> events = new ArrayList<>();
+        MultiPartyScenario scenario = getMultiPartyScenario(caseData);
+
+        if (defendant1ExtensionExists.test(caseData)) {
+            LocalDate deadlineExtension = caseData.getRespondentSolicitor1AgreedDeadlineExtension();
+            events.add(Event.builder()
+                               .eventSequence(prepareEventSequence(builder.build()))
+                               .eventCode(CONSENT_EXTENSION_FILING_DEFENCE.getCode())
+                               .dateReceived(caseData.getRespondent1TimeExtensionDate())
+                               .litigiousPartyID(RESPONDENT_ID)
+                               .eventDetailsText(getExtensionEventText(scenario,
+                                                                       deadlineExtension,
+                                                                       caseData.getRespondent1().getPartyName()))
+                               .eventDetails(EventDetails.builder()
+                                                 .agreedExtensionDate(deadlineExtension
+                                                                          .format(ISO_DATE))
+                                                 .build()).build());
         }
-        // date and time check to find the login
-        LocalDate extensionDate = caseData.getRespondentSolicitor1AgreedDeadlineExtension();
-
-        //finding extension date for the correct respondent in a 1v2 different solicitor scenario
-        MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
-        if (multiPartyScenario == ONE_V_TWO_TWO_LEGAL_REP) {
-            if ((caseData.getRespondent1TimeExtensionDate() == null)
-                && (caseData.getRespondent2TimeExtensionDate() != null)) {
-                extensionDate = caseData.getRespondentSolicitor2AgreedDeadlineExtension();
-                dateReceived  = caseData.getRespondent2TimeExtensionDate();
-            } else if ((caseData.getRespondent1TimeExtensionDate() != null)
-                && (caseData.getRespondent2TimeExtensionDate() != null)) {
-                if (caseData.getRespondent2TimeExtensionDate()
-                    .isAfter(caseData.getRespondent1TimeExtensionDate())) {
-                    extensionDate = caseData.getRespondentSolicitor2AgreedDeadlineExtension();
-                    dateReceived  = caseData.getRespondent2TimeExtensionDate();
-                } else {
-                    extensionDate = caseData.getRespondentSolicitor1AgreedDeadlineExtension();
-                    dateReceived  = caseData.getRespondent1TimeExtensionDate();
 
-                }
-            }
+        if (scenario == ONE_V_TWO_TWO_LEGAL_REP && defendant2ExtensionExists.test(caseData)) {
```

Comment:
this logic here will never be evaluated we may have to think about it and revise this code